src/Orchard.Web/Modules/Orchard.Users/Services/MembershipService.cs

```diff
@@ -118,7 +118,7 @@ public class MembershipService : IMembershipService {
                         template.Metadata.Wrappers.Add("Template_User_Wrapper");
 
                         var parameters = new Dictionary<string, object> {
-                            {"Subject", T("New account").Text},
+                            {"Subject", T("New account on "  + _orchardServices.WorkContext.CurrentSite.SiteName).Text},
```

Comment:
The correct way to do it is:
`T("New account on {0}.", _orchardServices.WorkContext.CurrentSite.SiteName).Text`

Otherwise you could not translate it.